plugins/rest/rest.go

```diff
@@ -124,7 +128,7 @@ func AuthPluginLookup(l func(string) HTTPAuthPlugin) func(*Client) {
 }
 
 // New returns a new Client for config.
-func New(config []byte, opts ...func(*Client)) (Client, error) {
+func New(config []byte, keys map[string]*keys.KeyConfig, opts ...func(*Client)) (Client, error) {
```

Comment:
I was wondering if there was a way to provide the keys via one of the `opts`:

I'd imagine something like
```go
func WithKeys(keys map[string]*keys.KeyConfig) func(*Client) {
    return func(c *Client) {
        c.keys = keys
    }
}
```

We wouldn't have to change the call sites to `rest.New()` then...

I might be missing something, you've added `keys` to `parsedConfig` (the `Config` struct)  instead for a reason surely...

Reply:
Ah, yeah that might be even better - will try it out :)

Reply:
Looking at this now and I'm not too sure about it. If we add a new `WithKeys` builder it would need to be called pretty much anywhere the client was created using `New`.. so the requirement itself is just as much a breaking change, it's only the signature of `New` that doesn't break, and arguably moving required config _out_ from the `New` constructor would be even more insidious as that requirement is not self-documenting. Or what would you say? 🤔

Reply:
💡 I wasn't aware that this was _required_ -- I had assumed only the rest plugins using JWT bearer grant_type would need keys. Sorry -- let's keep this as-is then 👍

Reply:
I think your assumption is correct for now (only JWT bearer auth requires keys), but the way the clients are instantiated (like [here](https://github.com/open-policy-agent/opa/blob/master/internal/config/config.go#L42)) I'm not sure how we'd differentiate a plugin that needs access to keys from one that does not. WDYT?